evalml/data_checks/data_check_message_code.py

```diff
@@ -32,6 +32,9 @@ class DataCheckMessageCode(Enum):
     TARGET_UNSUPPORTED_TYPE = "target_unsupported_type"
     """Message code for target data that is of an unsupported type."""
 
+    TARGET_REGRESSION_NON_NUMERIC = "target_regression_non_numeric"
```

Comment:
Maybe we just rename this to `TARGET_UNSUPPORTED_TYPE_REGRESSION`?  As named, it doesn't seem to get the point across to me.  Also a nit: "incompatible with regression"